include/SDL3/SDL_stdinc.h

```diff
@@ -1147,13 +1384,28 @@ extern SDL_DECLSPEC int SDLCALL SDL_wcscasecmp(const wchar_t *str1, const wchar_
  */
 extern SDL_DECLSPEC int SDLCALL SDL_wcsncasecmp(const wchar_t *str1, const wchar_t *str2, size_t maxlen);
 
-extern SDL_DECLSPEC long SDLCALL SDL_wcstol(const wchar_t *str, wchar_t **endp, int base);
+/**
+ * Convert a wide string to a long integer.
+ *
+ * `str` and `endp` must not overlap.
+ *
+ * \param str The null-terminated wide string to read. Must not be NULL and must not overlap with `endp`.
+ * \param endp If not NULL, the address of the first invalid character will be written. Must not overlap with `dst`.
+ * \param base The base of the integer to read. The values 0, 10 and 16 are supported.
+ *             If 0, the base will be inferred based on the integer's prefix.
+ * \returns The parsed long integer.
+ *
+ * \threadsafety It is safe to call this function from any thread.
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC long SDLCALL SDL_wcstol(const wchar_t * SDL_RESTRICT str, wchar_t ** SDL_RESTRICT endp, int base);
```

Comment:
I want to bring attention to the behavior of SDL's fallback implementations of `SDL_wcstol` and similar APIs, which I don't think match the behavior of most libc implementations. I didn't explicitly test them, but from just reading the code many of them appear to be subtly wrong. For example, `wcstol` supposed to support bases between 1 and 36, and specifying 0 should detect `0`-prefixed octal and `0x`/`0X`-prefixed, but SDL's implementation appears to only correctly support bases 1 to 10 and 16, and 0 only accepts lowercase `0x`.

Reply:
These mismatches between the libc functions and SDL's fallbacks make it difficult to correctly document these APIs, since it's not clear whether the difference in behavior is a bug or an intentional subset of the libc specs.

Reply:
Good catch!

This is a bug.
SDL's stdinc function aim to be a complete replacement of libc.
If you feel up to the job, a (failing) unit test in `test/testautomation_stdlib.c` would be appreciated.
And if you're still with us, you might be able to fix this.
(If you don't, we can jump in)

Reply:
I could try adding a few failing tests (if I do, should I disable them with `TEST_DISABLED` so that builds still pass?), but I don't think I would have the time/patience to fix all the behavior and and edge cases on my own. I could open a follow-up tracking issue instead. Especially with things like `SDL_strtod` and all the ways float parsing can go wrong it seems like it would require a quite significant amount of effort to achieve correctness.

Reply:
In general SDL's functions aim to be a functional subset of the C library functionality, not a complete implementation. In general we add what is commonly needed by games, and if it's missing an edge case or functionality we add it as needed by real-world use cases.

Reply:
Creating a new issue suffices :)

Reply:
#10750